### PR TITLE
Navigation: Next button behaviour 

### PR DESCRIPTION
**Pull request by Itxaka**
Uses the steps factory to track and set the current step finished to track if a step is finished or not and allow the next button to be enabled or disabled.

This is based on PR https://github.com/crowbar/crowbar-ui/pull/71 so it depends on that one to get merged.

### CODE REVIEW

**Comment by Itxaka** [comment]
[(3) S49P20 #6 Layout - Navigation bar](https://trello.com/c/LNr710jO/104-3-s49p20-6-layout-navigation-bar)

File: assets/app/features/upgrade/upgrade.controller.js [hunk 1]
```diff
@@ -3,53 +3,31 @@
...
             nextStep: nextStep,
-            isLastStep: isLastStep
+            isLastStep: isLastStep,
+            stepFinished: upgradeStepsFactory.stepFinished
         };
 
         // Get Steps list from provider
-        vm.steps.list = upgradeStepsFactory.getAll();
-        refeshStepsList();
+        vm.steps.list = upgradeStepsFactory.steps;
```
Comment (skazi0) [inline assets/app/features/upgrade/upgrade.controller.js:26]: move it above to `vm.steps` initialization just like `stepFinished`.

File: assets/app/data/crowbar/steps.factory.js [hunk 2]
```diff
@@ -4,72 +4,107 @@
...
+        function stepFinished() {
+            return factory.activeStep.finished;
+        }
+        function setCurrentStepCompleted() {
+            factory.activeStep.finished = true;
+        }
+
+        function refeshStepsList() {
+            var currentState = $state.current.name,
+                isCompletedStep = true;
```
Comment (skazi0) [inline assets/app/data/crowbar/steps.factory.js:29]: It took me some time to understand the logic of `isCompletedStep`. Not sure if there's a better name for this but maybe it should at least be described in a comment that this flag indicates that given step was already completed which will be reflected in the UI.
This is especially confusing and a bit redundant when connected with `setCurrentStepCompleted()` above.

File: assets/app/features/upgrade/upgrade.controller.js [hunk 3]
```diff
@@ -59,22 +37,16 @@
...
             }
-            vm.steps.activeStep.active = false;
-            vm.steps.activeStep.enabled = true;
-
-            vm.steps.activeStep = vm.steps.list[vm.steps.activeStep.id + 1];
-            vm.steps.activeStep.active = true;
-            vm.steps.activeStep.enabled = true;
 
-            $state.go(vm.steps.activeStep.state);
+            $state.go(vm.steps.list[upgradeStepsFactory.activeStep.id + 1].state);
```
Comment (skazi0) [inline assets/app/features/upgrade/upgrade.controller.js:41]: The whole `nextStep()` and `isLastStep()` belong in `upgradeStepsFactory`. The factory already uses `$state` and owns the steps list. They can be linked here just like the rest of `vm.steps`.

File: assets/app/features/upgrade/upgrade.controller.js [hunk 4]
```diff
@@ -3,53 +3,31 @@
...
-    function Upgrade7Controller($scope, $translate, $state, upgradeStepsFactory) {
+    function UpgradeController($scope, $translate, $state, upgradeStepsFactory) {
         var vm = this;
         vm.steps = {
             list: [],
-            activeStep: {},
             nextStep: nextStep,
-            isLastStep: isLastStep
+            isLastStep: isLastStep,
+            stepFinished: upgradeStepsFactory.stepFinished
```
Comment (skazi0) [inline assets/app/features/upgrade/upgrade.controller.js:22]: Not sure what our convention is but we should try to keep the functions returning booleans named consistently. Here we have `isLastStep()` and `stepFinished()`.

**Review by skazi0 — CHANGES_REQUESTED** [review]

File: assets/app/features/upgrade/upgrade.controller.js (hunk 3 above)
Comment (Itxaka) [reply assets/app/features/upgrade/upgrade.controller.js:41]: This actually comes from richa's PR, not this one. Stupid gituhb doent allow linked patches properly :/

File: assets/app/data/crowbar/steps.factory.js (hunk 2 above)
Comment (Itxaka) [reply assets/app/data/crowbar/steps.factory.js:29]: This actually comes from richa's PR, not this one. Stupid gituhb doent allow linked patches properly :/

Also I think that var is not used anywhere for anything at all and was already in the code, I guess for preparation for tracking the step completition, but Im not sure I understand the reasoning of it at all. Should be removed in future patches as I dont think is useful.

File: assets/app/features/upgrade/upgrade.controller.js (hunk 1 above)
Comment (Itxaka) [reply assets/app/features/upgrade/upgrade.controller.js:26]: This actually comes from richa's PR, not this one. Stupid gituhb doent allow linked patches properly :/

File: assets/app/data/crowbar/steps.factory.js (hunk 2 above)
Comment (santiph) [reply assets/app/data/crowbar/steps.factory.js:29]: This var was used in the prototype to mark the completed steps in green. See https://github.com/crowbar/crowbar-ui/blob/master/assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.jade#L2

It's a miss from the cleanup card and will be removed on Richa's changes

**Comment by Itxaka** [comment]
I kept breaking my git history while trying to rebase this properly, so I just extracted the commit, updated it and pushed a new PR: https://github.com/crowbar/crowbar-ui/pull/81

**Pull request closed without merging**